src/runtime/virtcontainers/qemu.go

```diff
@@ -882,20 +964,31 @@ func (q *qemu) StartVM(ctx context.Context, timeout int) error {
 
 	}
 
-	var strErr string
-	strErr, err = govmmQemu.LaunchQemu(q.qemuConfig, newQMPLogger())
+	qemuPid, err := govmmQemu.LaunchQemu(q.qemuConfig, newQMPLogger())
 	if err != nil {
-		if q.config.Debug && q.qemuConfig.LogFile != "" {
-			b, err := os.ReadFile(q.qemuConfig.LogFile)
-			if err == nil {
-				strErr += string(b)
-			}
+		q.Logger().WithError(err).Error("failed to launch qemu")
+		return fmt.Errorf("failed to launch qemu: %s", err)
+	}
+	if q.qemuConfig.Knobs.Daemonize {
+		// Wait for QEMU to daemonize itself. Don't check errors in
+		// case govmm has already reaped the upper process.
+		syscall.Wait4(qemuPid, nil, 0, nil)
```

Comment:
IIUC, the pid returned by `LaunchQemu` is the qemu command line process pid that is not the daemonized qemu main thread id. And the pid is already waited in `LaunchCustomQemu`.

Reply:
Same remark here : qemuPid corresponds to the top level QEMU process. We're just synchronizing with its termination to be sure that the daemonized QEMU is functional.